File: grader.rb
class Grader
    attr_accessor :student_name, :assignment_name, :score
    def initialize
        @student_name = "";
        @score = 0;
        @assignment_name = "";
    end

    def get_grade
        get_name
        get_assignment
        get_score
    end

    def get_name
        p "what is the student's name: ?"
        name = gets.chomp
        student_name = name
    end

    def get_assignment
        p "what is the assignment's name: ?"
        assignment = gets.chomp
        assignment_name = assignment
    end

    def get_score
        begin
            p "Enter student's score: ?"
            sc = gets.chomp.to_i
            score = sc
            grd = calculate_grade(sc)
            puts "Student Name: #{student_name}, Assignment: #{assignment_name}, score: #{sc}, Grade: #{grd} "
        rescue Exception => e   # storing the exception object in e
            puts e.message            # output error message
        end
    end


    private 


    def calculate_grade(s)
        begin
            # code at risk of failing here
            if(s >= 90 && s <= 100)
            
                return 'A';
            
            elsif(s >= 80 && s <= 89)
                return 'B';
            
            elsif(s >= 70 && s <= 79)
                return 'C';
            
            elsif(s >= 60 && s <= 69)
                return 'D';
            
            elsif(s < 60)
                return 'F';
            
            else
                raise Exception.new("Invalid score");
            end
        rescue Exception => e   # storing the exception object in e
            puts e.message            # output error message
        end
    end
end

g = Grader.new()
g.get_grade

File: grader.py
import sys

# Python Grader #

class Grader:
    name = ""
    assignment = ""
    def display_grade(self):
      self.get_name()
      self.get_assignment()
      self.get_score()
    def get_name(self):
      n = raw_input("what is the student's name: ? ")
      self.name = n
      
    def get_assignment(self):
      a = raw_input("what is the assignment's name: ? ")
      self.assignment = a

    def get_score(self):
      sc = int(raw_input("Enter student's score: ? "))
      grd = self.calculate_grade(sc)
      stuff_in_string = "Student Name: {}, Assignment: {}, score: {}, Grade: {}".format(self.name, self.assignment, sc, grd)
      print(stuff_in_string)
    
    def calculate_grade(self, s):
      try:
          if(s >= 90 and s <= 100):
              return 'A'
          elif(s >= 80 and s <= 89):
              return 'B'
          elif(s >= 70 and s <= 79):
              return 'C'
          elif(s >= 60 and s <= 69):
              return 'D'
          elif(s < 60):
              return 'F'
          else:
              raise ValueError("Invalid score")
      except ValueError:
          print("Invalid score")

p1 = Grader()
p1.display_grade()